Cython/Compiler/ExprNodes.py

```diff
@@ -2472,21 +2498,11 @@ def generate_deletion_code(self, code, ignore_nonexisting=False):
                 if self.cf_maybe_null and not ignore_nonexisting:
                     code.put_error_if_unbound(self.pos, self.entry)
 
-                if self.entry.type.is_pyobject:
-                    if self.entry.in_closure:
-                        # generator
-                        if ignore_nonexisting and self.cf_maybe_null:
-                            code.put_xgotref(self.result())
-                        else:
-                            code.put_gotref(self.result())
-                    if ignore_nonexisting and self.cf_maybe_null:
-                        code.put_xdecref(self.result(), self.ctype())
-                    else:
-                        code.put_decref(self.result(), self.ctype())
-                    code.putln('%s = NULL;' % self.result())
-                else:
-                    code.put_xdecref_memoryviewslice(self.entry.cname,
-                                                     have_gil=not self.nogil)
+                if self.entry.in_closure:
+                    # generator
+                    self.generate_gotref(code, handle_null=True, maybe_null_extra_check=ignore_nonexisting)
+                code.put_xdecref_clear(self.result(), self.ctype(),
```

Comment:
There used to be a decref (without x) here if `ignore_nonexisting and self.cf_maybe_null`. So this is a regression.

I don't care so much about giveref/gotref with or without x. They are just debug code that vanishes in production builds. I do care for incref/decref, though.